### ISSUE TITLE
Support for Array fields like uuid[] on Postgres

### ISSUE THREAD

**Issue by jlvertol** [com-lihaoyi/scalasql#40]
Hi! As far as I could tell while using Scalasql with Postgres, SQL array columns are not supported.

I tried to add a property of type `T[Seq[String]]` to my table, but got this error:

```
[error] 109 |object Invoice extends Table[Invoice]:
[error]     |                                     ^
[error]     |No given instance of type scalasql.core.Queryable.Row[scalasql.core.Expr[Seq[String]], Seq[String]] was found.
[error]     |I found:
[error]     |
[error]     |    scalasql.core.Expr.ExprQueryable[scalasql.core.Expr, Seq[String]](
[error]     |      scalasql.core.TypeMapper.enumTypeFromDialectTypeMappers[T])
[error]     |
[error]     |But method enumTypeFromDialectTypeMappers in object TypeMapper does not match type scalasql.core.TypeMapper[Seq[String]].
```

Additionally, if I just try to use `T[String]` as a workaround, I get a different error when querying the table instead:

```
org.postgresql.util.PSQLException: ERROR: operator does not exist: uuid[] ~~ text
  Hint: No operator matches the given name and argument types. You might need to add explicit type casts.
```

Am I correct that we don't support these types?

If so, I'd like to try my hand at solving it, but first I want to make sure this is actually not supported and there's no workarounds.

**Comment by lihaoyi** [comment]
Yes they're not supported right now, would love to have your help in adding support!

**Comment by NovaMage** [comment]
@jlvertol Did you end up taking a stab at this or did you find a workaround?  This is the only thing stopping me from adopting this library at my company.

**Comment by jlvertol** [comment]
> [@jlvertol](https://github.com/jlvertol) Did you end up taking a stab at this or did you find a workaround? This is the only thing stopping me from adopting this library at my company.

Hey @NovaMage, I did take a stab at it, but given time constraints it was only good enough to fix it for me and my company (by making a workaround class in our own repository). The upside is that Li's library is indeed flexible enough to allow such workarounds; you don't even need to fork ScalaSQL, it's enough to make an implicit class for the kind of Arrays you need to support.

As for the patch, I don't know when I'll be able to come back and make a patch for this.
